### DISCUSSION TITLE
[General] Running local network with zksync-cli and foundry-zksync produces following error: Transaction HALT: Account validation error: Account validation returned invalid magic value.

### DISCUSSION THREAD

**Discussion by MajorDomDePIN** [zkSync-Community-Hub/zksync-developers#647] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hi all,

I've been using Cyfrin's Foundry Fundamentals course (https://updraft.cyfrin.io/courses/foundry/foundry-simple-storage/zksync-local-deploy) for a few days now to improve my knowledge of Foundry. 

A few days ago I was already able to deploy to the local zksync-chain (zksync-cli) using foundry, as described in the course. 

Today I wanted to continue at this point in the course and therefore start the Docker node for the second time and interact with it. The first time this worked as described without any problems. Now, depending on which zksync private key I use, I get the following output:

command:
forge create ./src/SimpleStorage.sol:SimpleStorage --private-key 0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356 --rpc-url http://127.0.0.1:8011/ --legacy --zksync
Output:
Server has returned an error response: Error code 3: Execution error: transaction HALT: account validation error: insufficient funds for charge + value

Command:
forge create ./src/SimpleStorage.sol:SimpleStorage --rpc-url http://127.0.0.1:8011/ --private-key 0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 --legacy --zksync
Output:
Server has returned an error response: Error code 3: Execution error: transaction HALT: Account validation error: account validation returned an invalid magic value. Most of the time this means that the signature is incorrect.

I am working with foundryup-zksync and just don't know why this problem is occurring. If anyone can help me with this I would be more than happy

**Comment by alhenaai** [comment]
The error message you're encountering, "Account validation error: account validation returned an invalid magic value," typically indicates an issue with the account validation process during the transaction. This can often be related to the signature or the setup of the account you are using.

Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check the Magic Value Requirement**:
   - Ensure that the account and paymasters are returning the correct magic value upon validation. This is crucial for transactions to be accepted on the mainnet. More details on this can be found in the [ZKsync documentation on the returned magic value](https://docs.zksync.io/zk-stack/components/zksync-evm/bootloader#returned-magic-value).

2. **Verify Account Setup**:
   - Double-check the setup of your account, especially the private keys and their corresponding public addresses. Make sure that the accounts are correctly configured and funded for the transactions you are attempting.

3. **Update and Verify Tools**:
   - Ensure that your `zksync-cli` and `foundry-zksync` tools are up to date. This can sometimes resolve compatibility issues that might be causing the error.

4. **Consult Documentation and Community**:
   - Review the relevant sections of the ZKsync documentation and consider reaching out to the community or support if the problem persists. The community might have encountered and resolved similar issues.

If these steps do not resolve the issue, it might be beneficial to provide more detailed logs or error messages to the ZKsync support team or community for further assistance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
@MajorDomDePIN what did you do to get this resolved?

**Comment by MajorDomDePIN** [comment]
Thank you for your Answer!

I recloned the project folder into my wsl so i only working inside of it and not in windows anymore, but this couldn't solve the problem. 

i tried to run the container for the zksync-docker-machine in wsl as well as in windows, but both times i had the same issues with both of them.

currently i only can deploy with the first private key which is listed on https://docs.zksync.io/build/test-and-debug/in-memory-node, if i try the second one instead i got a "not enought balance" error...
![grafik](https://github.com/user-attachments/assets/51c55ac4-5d4b-4d3a-9663-a41868247fbd)

Unfortunately, these private keys do not match the private keys that are listed in the docker container. 
![grafik](https://github.com/user-attachments/assets/b4dc4ec5-3553-422a-abd8-209972928e40)

Regarding the keys from the container, they seem to work completely arbitrarily or randomly

![grafik](https://github.com/user-attachments/assets/743058fc-2124-4974-bb45-0171d8779914)
![grafik](https://github.com/user-attachments/assets/bdf7630c-7d4c-479a-a824-50ddb1b0f97b)

here is a script i made to try to deploy to every private key from the domain as well as the local node:
https://github.com/MajorDomDePIN/Deployment_Test

And if a transaction goes through, no broadcast file is created for chain 260, as already mentioned

**Reply by bxpana** [reply]
It seems like you had marked the discussion as closed, but to confirm you're still having issues with deploying contracts to the In-Memory Node on your Windows using WSL?
